glib/src/error.rs

```diff
@@ -206,6 +213,25 @@ impl error::Error for BoolError {}
 
 #[cfg(test)]
 mod tests {
+    use super::*;
+    use std::ffi::CString;
+
+    #[test]
+    fn test_into_raw() {
+        let e = Error::new(crate::FileError::Failed, "Failed").into_raw();
+        unsafe {
+            assert_eq!((*e).domain, ffi::g_file_error_quark());
+            assert_eq!((*e).code, ffi::G_FILE_ERROR_FAILED);
+            assert_eq!(
+                CString::from_raw((*e).message),
+                CString::new("Failed").unwrap()
```

Comment:
```suggestion
                CStr::from_ptr((*e).message),
                CString::new("Failed").unwrap().as_c_str()
```

Reply:
You were stealing the message from the error here, and that's why it failed freeing below with a double free.

Reply:
It is *always* my fault :)